src/components/App/index.tsx

```diff
@@ -0,0 +1,329 @@
+              with your improvements. Thank you and enjoy!
+            </p>
+          </div>
+        )
+      });
+
+      Cookies.set(cookieKey, 'true', { expires: 365 });
+    }
+  }, []);
+
+  // Fetch the current term's scraper information
+  useEffect(() => {
+    setOscar(null);
+    if (term) {
+      axios
+        .get(`https://gt-scheduler.github.io/crawler/${term}.json`)
+        .then((res) => {
+          const newOscar = new Oscar(res.data);
+          setOscar(newOscar);
+        });
+    }
+  }, [term]);
+
+  // Fetch all terms via the GitHub API
+  useEffect(() => {
+    axios
+      .get(
+        'https://api.github.com/repos/gt-scheduler/crawler/contents?ref=gh-pages'
+      )
+      .then((res) => {
+        const newTerms = (res.data as { name: string }[])
+          .map((content) => content.name)
+          .filter((name: string) => /\d{6}\.json/.test(name))
+          .map((name: string) => name.replace(/\.json$/, ''))
+          .sort()
+          .reverse();
+        setTerms(newTerms);
+      });
+  }, [setTerms]);
+
+  // Set the term to be the first one if it is unset
+  // (once the terms load)
+  useEffect(() => {
+    const term_from_cookie = Cookies.get('term');
+    if (!term && (!term_from_cookie || term_from_cookie === 'undefined')) {
+      const [recentTerm] = terms;
+      setTerm(recentTerm);
+    } else if (term_from_cookie && term_from_cookie !== 'undefined') {
+      setTerm(term_from_cookie);
+    }
+  }, [terms, term, setTerm]);
+
+  // Initialize the versionName to Primary
+  useEffect(() => {
+    const v = Cookies.get('version');
+    if (!versionName) {
+      setVersionName(v || 'Primary');
+    }
+  }, [versionName, setVersionName]);
+
+  // Initialize the version lists for each term
+  useEffect(() => {
+    const vs = Cookies.get('versions');
+    if (!vs || vs === '{}') {
+      patchVersionsData(
+        terms.reduce((ac, cur) => ({ ...ac, [cur]: ['Primary', 'New'] }), {})
+      );
+    }
+  }, [patchVersionsData, terms]);
+
+  // Check for new terms
+  // if there are new terms, initialize new version lists
+  useEffect(() => {
+    if (typeof versionLists === 'object' && terms.length !== 0) {
+      const terms_in_cookie = Object.keys(versionLists);
+      if (terms_in_cookie.length !== terms.length) {
+        const missing_terms = terms.filter((x) => !terms_in_cookie.includes(x));
+        patchVersionsData(
+          Object.assign(
+            versionLists,
+            missing_terms.reduce(
+              (ac, cur) => ({ ...ac, [cur]: ['Primary', 'New'] }),
+              {}
+            )
+          )
+        );
+      }
+    }
+  }, [patchVersionsData, terms, versionLists]);
+
+  // Backward compatibility for schedule data before version switch implementation
+  // Copy the old cookies to match with the new cookies' naming format.
+  // Checks for term 202008, 202102, 202105, 202108 at the time of implementation.
+  // Does not remove the old cookies.
+  useEffect(() => {
```

Comment:
This is good, and I appreciate the comment to describe where the magic numbers come from. Just to clarify: this works because "Primary" is _always_ included as a key in the `versionLists`, right (so the `versionLists` cookie itself doesn't have to be modified in this code)?

Reply:
Yes, the initialization of `versionLists` adds `Primary` to the version list of each term. When user switch to the new version, there will be no cookie storing verison lists; the initialization automatically adds in `Primary`'s to the cookie.